### PR TITLE
Functionality to leave matchmaking

### PR DESCRIPTION
**Pull request by AminArria**
Closes #[issue]

## Motivation

Allow player to leave matchmaking

## Summary of changes

- [x] Add functionality to leave matchmaking
- [x] Button in UI to trigger leave
- [x] Refactor a bit lobby websocket messages, now we have more and they fall under `LobbyEvent` 

## How has this been tested?

Test with https://github.com/lambdaclass/mirra_backend/pull/456

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Comment by AngieDutra** [comment]
Nice! I'll review this now

**Comment by AngieDutra** [comment]
Client seems to work but I commented a bug in backend PR

**Review by AngieDutra — APPROVED** [review]
Awesome. Tested on Android build ✨

**Review by alfopisano — DISMISSED** [review]
Nice!! Tested on iOS build

**Review by Nico-Sanchez — DISMISSED** [review]

**Comment by AngieDutra** [comment]
I had a weird thing happen. I was playing with Android build and Unity desktop. I canceled the first match with desktop and everything worked fine. But for the second match I wasn't able to join with my Unity desktop. It got stuck in the loading screen of prepare for battle. The fight image. However, I was able to see my desktop player in my game

File: client/Assets/Scripts/LobbyManager.cs [hunk 1]
```diff
@@ -57,4 +57,11 @@ public IEnumerator WaitForLobbyJoin()
         );
         SceneManager.LoadScene(BATTLE_SCENE_NAME);
     }
+
+    public void LeaveLobby(){
+        ServerConnection.Instance.LeaveLobby();
+        if(System.String.IsNullOrEmpty(ServerConnection.Instance.GameSession)){
```
Comment (manucamejo) [inline client/Assets/Scripts/LobbyManager.cs:63]: I'm not sure if this is correct; perhaps we should wait for the server's response before sending the player to the MainScreen.

**Review by manucamejo — CHANGES_REQUESTED** [review]

File: client/Assets/Scripts/LobbyManager.cs (hunk 1 above)
Comment (AminArria) [reply client/Assets/Scripts/LobbyManager.cs]: Done, now we receive a message for asking to leave and the response for it

**Review by agustinesco — APPROVED** [review]
LGTM! tested in both the unity client and in an android build and worked as expected!

this was recorded in android:
https://github.com/lambdaclass/curse_of_mirra/assets/106101218/ee3c3eed-7eb0-4ce9-98d3-aad0333b9f4f

**Comment by AngieDutra** [comment]
I think the quick game button is broken in this PR

**Review by AngieDutra — APPROVED** [review]
Tested on Android build and is working as expected.
In the future re-design I would notice that the cancel button is in a similar position than the play button and this may not be wat we want in our UX

**Pull request merged**